### PR TITLE
fix: better elixir docs and defaults

### PR DESCRIPTION
**Pull request by iloveitaly**
Two main improvements:

- **docs: improve elixir build documentation**
- **fix: better elixir defaults, missing packages**

I don't think additional tests are needed here. Refactored to pattern off of the python provider a bit more closely.

### CODE REVIEW

**Comment by iloveitaly** [comment]
Friendly reminder regarding this pull request! Please let me know if there are any further updates needed.

File: src/providers/elixir.rs [hunk 1]
```diff
@@ -51,9 +45,6 @@ impl Provider for ElixirProvider {
             build_phase.add_cmd("mix assets.deploy".to_string());
         }
 
-        if mix_exs_content.contains("postgrex") && mix_exs_content.contains("ecto") {
-            build_phase.add_cmd("mix ecto.setup");
```
Comment (coffee-cup) [inline src/providers/elixir.rs:55]: Why was this removed?

**Review by coffee-cup — COMMENTED** [review]
Thanks for the PR! Sorry its taken so long to get to and review.

I've updated the snapshot tests for elixir that was failing the build. Just one comment about the ecto.setup being removed, but besides that, it looks good to go.

File: src/providers/elixir.rs (hunk 1 above)
Comment (iloveitaly) [reply src/providers/elixir.rs:55]: Ah, for me I didn't want this. Will add it back in.

The risk with running ecto.setup is it generally requires a DB connection, which is not available when the image is being built.

**Review by coffee-cup — APPROVED** [review]
